File: POO/Exo3/implementation Java/ListEntity.java
public class ListEntity
{
    public Cat listCat[]= new Cat[5];
    public Mouse listMouse[]= new Mouse[10];
    public Maize listMaize[]= new Maize[8];
    public int nbCat, nbMouse, nbMaize;

    public ListEntity()
    {
        this.nbCat=0;
        this.nbMouse=0;
        this.nbMaize=0;
    }

    public void AddCat(Cat elt)
	{
		this.nbCat++;
		this.listCat[nbCat]=elt;
	}
	public void AddMouse(Mouse elt)
	{
		this.nbMouse++;
		this.listMouse[nbMouse]=elt;
	}
	public void AddMaize(Maize elt)
	{
		this.nbMaize++;
		this.listMaize[nbMaize]=elt;
	}
	public void rmCat()
	{
		this.nbCat--;
	}
	public void rmMouse()
	{
		this.nbMouse++;
	}
	public void rmMaize()
	{
		this.nbMaize--;
	}
}

File: POO/Exo3/Implentation C++/ListEntity.cpp
#include <iostream>
#include "ListEntity.hpp"
#include "Cat.hpp"
#include "Mouse.hpp"
#include "Maize.hpp"


using namespace std;

ListEntity:ListEntity()
	{
		nbMaize=0;
		nbMouse=0;
		nbCat=0;
	}
	void ListEntity::AddCat(Cat elt)
	{
		nbCat++;
		listCat[nbCat]=elt;
	}
	void ListEntity::AddMouse(Mouse elt)
	{
		nbMouse++;
		listMouse[nbMouse]=elt;
	}
	void ListEntity::AddMaize(Maize elt)
	{
		nbMaize++;
		listMaize[nbMaize]=elt;
	}
	void ListEntity::rmCat()
	{
		nbCat--;
	}
	void ListEntity::rmMouse()
	{
		nbMouse++;
	}
	void ListEntity::rmMaize()
	{
		nbMaize--;
	}